GetCertificate/handler.ts

```diff
@@ -102,7 +104,55 @@ export const GetCertificateHandler = (
     TE.chain(e => {
       switch (e.status) {
         case 200:
-          return TE.of(e.value);
+        case 404:
+          return pipe(
```

Comment:
Readability note: consider move this sub procedure into a dedicate, well-named function